gnomad_qc/v3/create_release/make_var_annot_hists.py

```diff
@@ -1,71 +1,173 @@
 import argparse
+import json
+import logging
 
 import hail as hl
-from gnomad.utils.annotations import (ANNOTATIONS_HISTS,
-                                      create_frequency_bins_expr,
-                                      get_annotations_hists)
-from gnomad.utils.slack import try_slack
+from gnomad.resources.resource_utils import DataException
+from gnomad.utils.annotations import create_frequency_bins_expr, get_annotations_hists
+from gnomad.utils.file_utils import file_exists
+from gnomad.utils.slack import slack_notifications
 
-from gnomad_qc.v3.resources.constants import CURRENT_RELEASE
-from gnomad_qc.v3.resources.release import (qual_hists_json_path,
-                                            release_ht_path)
+from gnomad_qc.slack_creds import slack_token
+from gnomad_qc.v3.resources.release import (
+    annotation_hists_path,
+    qual_hists_json_path,
+    release_ht_path,
+)
 
 
-def main(args):
-    hl.init(default_reference='GRCh38', log='/variant_histograms.log')
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+LOG10_ANNOTATIONS = ["AS_VarDP", "QUALapprox", "AS_QUALapprox"]
+"""
+List of annotations to log scale when creating histograms. 
+"""
+
 
-    ht = hl.read_table(release_ht_path())
-    # NOTE: histogram aggregations are done on the entire callset (not just PASS variants), on raw data
+def create_frequency_bins_expr_inbreeding(
+    AF: hl.expr.NumericExpression,
+) -> hl.expr.StringExpression:
+    """
+    Creates bins for frequencies in preparation for aggregating QUAL by frequency bin.
 
-    hist_dict = ANNOTATIONS_HISTS
-    hist_dict['MQ'] = (20, 60, 40) # Boundaries changed for v3, but could be a good idea to settle on a standard
-    hist_ranges_expr = get_annotations_hists(
-        ht,
-        ANNOTATIONS_HISTS
+    Uses bins of < 0.0005 and >= 0.0005
+
+    NOTE: Frequencies should be frequencies from raw data.
+    Used when creating site quality distribution json files.
+
+    :param AF: Field in input that contains the allele frequency information
+    :return: Expression containing bin name
+    :rtype: hl.expr.StringExpression
+    """
+    bin_expr = (
+        hl.case()
+        .when(AF < 0.0005, "under_0.0005")
+        .when(AF >= 0.0005, "over_0.0005")
+        .or_missing()
     )
+    return bin_expr
+
+
+def main(args):
+    hl.init(default_reference="GRCh38", log="/variant_histograms.log")
+
+    logger.info("Loading ANNOTATIONS_HISTS dictionary...")
+    if not file_exists(annotation_hists_path()):
+        raise DataException(
+            "Annotation hists JSON file not found. Need to create this JSON before running script!"
+        )
+
+    with hl.hadoop_open(annotation_hists_path()) as a:
+        ANNOTATIONS_HISTS = json.loads(a.read())
```

Comment:
Is `ANNOTATIONS` in `ANNOTATIONS_HISTS` supposed to be plural while it's singular `annotation` in `annotation_hists_path()`?

Reply:
This is imported from `gnomad_methods` and I think used in other repos so I think I will leave `ANNOTATIONS_HISTS` I think it should probably be singular though